### PR TITLE
FLAMMING B52

### PR DESCRIPTION
**Pull request by DeityLink**
- FLAMMING B52

b52

Bartender was my first job when I started playing SS13. It was about damn time that I showed some love for the job as a coder.

### CODE REVIEW

**Comment by Probe1** [comment]
he didn't play engineer master race as his first job 

what about in hand lighting?

**Comment by DeityLink** [comment]
nope, these aren't flashlights, they'll only light stuff when they are on the ground.

File: code/modules/reagents/reagent_containers/food/drinks.dm [hunk 1]
```diff
@@ -1173,50 +1179,60 @@
 	else if(istype(I, /obj/item/weapon/weldingtool))
```
Comment (ComicIronic) [inline code/modules/reagents/reagent_containers/food/drinks.dm]: Replace all of this block with an is_hot() check instead.

File: code/modules/reagents/reagent_containers/food/drinks.dm [hunk 2]
```diff
@@ -1173,50 +1179,60 @@
...
+	var/flavor_text = "<span  class='rose'>[user] lights \the [name] with \the [I].</span>"
 	if(!lit && molotov == 1)
 		lit = 1
 		for(var/mob/O in viewers(usr, null))
 			O.show_message(flavor_text, 1)
 		processing_objects.Add(src)
 		update_icon()
+	if(!lit && flammable)
+		lit = 1
+		for(var/mob/O in viewers(usr, null))
```
Comment (ComicIronic) [inline code/modules/reagents/reagent_containers/food/drinks.dm]: AUGH, NO.
Use visible_message

Comment (DeityLink) [inline code/modules/reagents/reagent_containers/food/drinks.dm]: sure, but remember to blame whoever coded those molotovs in the first place.

Comment (ComicIronic) [inline code/modules/reagents/reagent_containers/food/drinks.dm]: I still blame you for copypaste

**Comment by Probe1** [comment]
I think they'd be cooler if they had an in hand illumination sprite but all the same they're a good addition

File: code/modules/reagents/reagent_containers/food/drinks.dm (hunk 1 above)
Comment (DeityLink) [inline code/modules/reagents/reagent_containers/food/drinks.dm]: We have an is_hot proc? well that changes everything.

**Comment by DeityLink** [comment]
@Probe1 I'm pretty sure that some other coder was working on that, I've seen a gif in the thread of it a while ago.

File: code/modules/reagents/reagent_containers/food/drinks.dm [hunk 3]
```diff
@@ -264,9 +267,12 @@
 
 /obj/item/weapon/reagent_containers/food/drinks/proc/imbibe(mob/user) //Drink the liquid within
 
-	user << "<span  class='notice'>You swallow a gulp of \the [src].</span>"
+	user << "<span  class='notice'>You swallow a gulp of \the [src].[lit ? " It's hot!" : ""]</span>"
 	playsound(user.loc,'sound/items/drink.ogg', rand(10,50), 1)
 
+	if(lit)
```
Comment (ComicIronic) [inline code/modules/reagents/reagent_containers/food/drinks.dm]: There's been issues recently with silicons taking damage from manual temperature increases.
I think you should do
`if(lit && !issilicon(user))`
And we'll replace it with a mob proc at a later time.

**Comment by DeityLink** [comment]
Or I just might fix those silicon temperature issues. I've been thinking if giving them a "cooler" component.

**Comment by DeityLink** [comment]
fixed conflicts. that'd be nice if PRs that need final decision got their verdict before new conflicts appear >__>

**Pull request merged**